crypto/x509/v3_purp.c

```diff
@@ -18,26 +18,26 @@
 
 static int check_ssl_ca(const X509 *x);
 static int check_purpose_ssl_client(const X509_PURPOSE *xp, const X509 *x,
-                                    int require_ca);
+                                    int non_leaf);
```

Comment:
I am not sure I like this change. `require_ca` might be misleading but something like `must_be_ca` should fix that. `non_leaf` IMO actually hides what the parameter is about.

Reply:
But thinking about it more yeah, `non_leaf` is OK for these functions.